src/kv/store.h

```diff
@@ -997,6 +997,19 @@ namespace kv
      **/
     void swap_private_maps(Store& store)
     {
+      {
+        const auto source_version = store.current_version();
+        const auto target_version = current_version();
+        if (source_version != target_version)
```

Comment:
I feel like this check will break recovery where we have the CCF KV (public-only at this point, i.e. private maps are empty) and the recovery KV (both private and public). I don't think that the two stores are guaranteed to be at the same version when the swap is performed.

Reply:
Yeah that's definitely not guaranteed, the public store is likely further ahead, which is fine (it's guaranteed not have any private state after the version the private store reached when it finished recovery).

Reply:
This check is now `if (source_version > target_version)` - we always expect to copy private maps _from an earlier (or equal) version_, meaning the versions in the rolls we swap in should correctly be lower than the target's current version.